types/model/name.go

```diff
@@ -290,12 +290,14 @@ func (n Name) Filepath() string {
 	if !n.IsFullyQualified() {
 		panic("illegal attempt to get filepath of invalid name")
 	}
-	return strings.ToLower(filepath.Join(
-		n.Host,
-		n.Namespace,
-		n.Model,
+	return filepath.Join(
+		strings.ToLower(filepath.Join(
+			n.Host,
+			n.Namespace,
+			n.Model,
+		)),
 		n.Tag,
```

Comment:
Omitting Tag from normalization means we now have another class of bugs awaiting. This is just sweeping the problem from one rug to another. 

If Tag should not be lower, then make it upper, but leaving it as-is means that user input could be wrong and we see the same problem, but in a different light.

Also: The ollama subcommands with code paths going to disk could use more tests to catch things like this sooner.